### ISSUE TITLE
Ombi is not marking TV series as available/Requests are not added to Sonarr

### ISSUE THREAD

**Issue by SpookyRoger** [tidusjar/Ombi#3282]
**Describe the bug**
TV series are not marked as available.  When a requests for an available show is made it shows "request added we already have this request, please choose the Select option to refine your request"

TV series requests made by an auto approve tv/movie user are not added to sonarr.

**To Reproduce**
--

**Expected behavior**
Expecting ombi to pick up my library and mark shows as available, requests made to be added to sonarr.

**Screenshots**
https://postimg.cc/gallery/3e8mb9096/

**Logs (Logs directory where Ombi is located)**
If applicable, a snippet of the logs that seems relevant to the bug if present.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

**Ombi Version (please complete the following information):**
 - Version 	3.0.4900 
- Media Server: Plex

**Additional context**
Screenshots posted above showing the problem and also screenshot of the log at the time of the Sonarr sync job.

**Comment by SpookyRoger** [comment]
![plex agent](https://user-images.githubusercontent.com/18377193/69489326-0ef84600-0edb-11ea-8e64-6f884eb3a33f.PNG)
![plexagent2](https://user-images.githubusercontent.com/18377193/69489334-3222f580-0edb-11ea-82b9-bdd762b87181.PNG)

additional screenshots regarding the FAQ entry about TV series availability.

**Comment by tidusjar** [comment]
Thanks for the report. 

Can you press the select... Button on the TV show and then it well tell you what episodes ombi thinks are missing

**Comment by SpookyRoger** [comment]
All episodes show "processing request"

**Comment by tidusjar** [comment]
Is ombi connected to your Plex server correctly?

**Comment by SpookyRoger** [comment]
I believe so
![Ombi5](https://user-images.githubusercontent.com/18377193/69505280-fa28ba80-0f8d-11ea-8870-1295e29519a7.PNG)

**Comment by SpookyRoger** [comment]
I believe so, Test connectivity says "successfully connected"

**Comment by tidusjar** [comment]
Ok press the manual sync button and then upload the log file from that sync please

**Comment by SpookyRoger** [comment]
[New Text Document.txt](https://github.com/tidusjar/Ombi/files/3886068/New.Text.Document.txt)


Hopefully this is what you are looking for. Tried my best.

**Comment by leloberg** [comment]
I'm getting the same issue at my end. Same log too. I'm running the Ombi docker container, if that is important. I've tried to restart ombi, without luck. Connectivity to Plex is working. It seems to just be an issue with TV-shows. As for movies it's also catching movies added directly from Radarr or movies found in Plex.

**Comment by trevor-s8** [comment]
Same issue here. Running as a docker in Unraid.

I also had the movie add SQL bug and fixed that by manually adding a column to the table.

**Comment by TheCreatorzOne** [comment]
Same issue here as well, all connections are fine but TV Shows need to manually be marked as available while Movies are detected just fine :/

**Comment by chadillac0121** [comment]
Anyone have an update on this issue?

Me and a friend of mine both just noticed this exact same issue. All TV shows, even ones where all episodes exist, show "Request" in Ombi. If selecting specific episodes, every episode for every show is marked at "Processing Request", making it impossible to request TV shows in Ombi. This previously worked fine for both of us.

Connectivity works fine, all movies have loaded in properly.

**Comment by kimpenhaus** [comment]
Not sure how you guys handle your ombi / docker instances? Are they shut down during night times? I was facing the exact same situation that all my shows were shown as processing request - I then realized that the ombi cron for plex sync was during night hours when my docker host was shut down. I changed the plex sync cron to fire during day hours and now all is good again.

Maybe you can double check if that meets your setup as well.

**Comment by dannyhanes** [comment]
I am also having this same issue. Everything is is showing connected (Plex, Sonarr v3, Radarr). I have the v3 box checked for sonarr, and all of my profiles, languages etc are populating in the settings. Plex is showing as connected as well. However none of my TV shows are showing as available.

**Comment by damusmf** [comment]
Any updates on this?  Running in to the same issue on both v3 and v4 of ombi.  Tried everything including a full database wipe and set everything back up again.  Most TV shows are not showing as available still.  Connected to sonarr v3 and plex fine.  Movies seem to work, no errors in the logs.  Just not marking shows as available.

**Comment by tidusjar** [comment]
Would be worth looking inside your Database (Sqlite = OmbiExternal.db, MySql = Whatever you setup for the external database) and query the SonarrCache table to see what data is present in there.

**Comment by dannyhanes** [comment]
@tidusjar I just took a look at mine, and when compared to Radarr (which appears to be working) there is no column for "HasFile".

![Screen Shot 2020-03-02 at 1 08 48 PM](https://user-images.githubusercontent.com/3778687/75704233-fa201100-5c86-11ea-876b-686ad766a7af.png)

**Comment by tidusjar** [comment]
What about the sonarrepisodecache ?

**Comment by dannyhanes** [comment]
Attached

![Screen Shot 2020-03-02 at 4 01 19 PM](https://user-images.githubusercontent.com/3778687/75717415-14fe7f80-5c9f-11ea-9661-b4a32d2392c5.png)

**Comment by tidusjar** [comment]
Ok so it looks like the sonarr sync is working. 
Now can you check the Plex episode table for parks and rec season 5 ep 13?

**Comment by dannyhanes** [comment]
Going to throw a few screenshots in here that might also help. It looks 98% of my shows are stuck as processing request. 

PlexEpisode
![Screen Shot 2020-03-02 at 4 11 11 PM](https://user-images.githubusercontent.com/3778687/75718309-b2a67e80-5ca0-11ea-901f-b9f226266831.png)

Ombi - Request Specific Season
<img width="1208" alt="Screen Shot 2020-03-02 at 4 09 06 PM" src="https://user-images.githubusercontent.com/3778687/75718320-b89c5f80-5ca0-11ea-8967-0149ec797469.png">

Existing File in Plex
<img width="753" alt="Screen Shot 2020-03-02 at 4 09 45 PM" src="https://user-images.githubusercontent.com/3778687/75718340-c2be5e00-5ca0-11ea-921c-9da44b3434e7.png">

**Comment by absentbri** [comment]
I'm having the same issue with UnRaid using dockers (all provided by linuxserverio), everything is connecting fine, requests fire off everything to Sonarr/Radarr. Movies working great, very happy with them.

But TV Shows are not being synced at all. Even though in the docker log it's showing its found them and processing them. It's so strange...

Excerpt of log while running "Manually Run Full Sync"
[log.txt](https://github.com/tidusjar/Ombi/files/4364628/log.txt)

**Comment by tatzu** [comment]
same issue her also runing it thrr plexguide,
Movies is working fine but series isnt showing as avalibile and cant request

**Comment by jes1417** [comment]
@tatzu same issues here, requests not Being sent to sonarr I posted in [#3434](https://github.com/tidusjar/Ombi/issues/3434)

Sonarr is saying

```
DiskScanServiceScanning Beverly Hills, 902102:35pmRefreshSeriesServiceSkipping refresh of series: Beverly Hills, 902102:35pmExistingExtraFileServiceFound 0 extra files2:35pmExistingOtherExtraImporterFound 0 existing other extra files2:35pmExistingSubtitleImporterFound 0 existing subtitle files2:35pmExistingMetadataImporterFound 0 existing metadata files2:35pmDiskScanServiceCompleted scanning disk for Belgravia2:35pmDiskScanServiceScanning Belgravia2:35pmRefreshEpisodeServiceFinished episode refresh for series: [362204][Belgravia].2:35pmRefreshEpisodeServiceStarting episode info refresh for: [362204][Belgravia]2:35pmRefreshSeriesServiceUpdating Belgravia2:35pmExistingExtraFileServiceFound 0 extra files2:35pmExistingOtherExtraImporterFound 0 existing other extra files2:35pmExistingSubtitleImporterFound 0 existing subtitle files2:35pmExistingMetadataImporterFound 0 existing metadata files2:35pmDiskScanServiceCompleted scanning disk for The Adventures of Super Mario Bros. 32:35pmDiskScanServiceScanning The Adventures of Super Mario Bros. 32:35pmRefreshSeriesServiceSkipping refresh of series: The Adventures of Super Mario Bros. 32:35pmRssSyncServiceRSS Sync Completed. Reports found: 226, Reports grabbed: 02:23pmDownloadDecisionMakerProcessing 226 releases2:23pmRssSyncServiceStarting RSS Sync2:23pmRssSyncServiceRSS Sync Completed. Reports found: 226, Reports grabbed: 02:08pmDownloadDecisionMakerProcessing 226 releases2:08pmRssSyncServiceStarting RSS Sync2:08pmSonarrErrorPipelineAPI Error: BadRequest: 0 is not a valid ID2:04pmSonarrErrorPipelineInvalid request Validation failed: -- Path: Invalid Path -- RootFolderPath: Invalid Path2:04pmSonarrErrorPipelineInvalid request Validation failed: -- Path: Invalid Path -- RootFolderPath: Invalid Path
```

But everything is correct it all used to work

**Comment by bugsyb** [comment]
+1 
same issue here Ombi version: 3.0.4892 (docker based)
Requests work fine, Sonarr finishes download and shows as available yet nothing is updated on Ombi side and it doesn't get marked as Available.

Thought - could it be that Jellyfin renames file/folder? But that happens potentially (not confirmed) only after Sonarr finishes the job and that is where it is marked as finished and as such Ombi should be able to get that discovered properly?

How to call Ombi/Sonarr sync manually to force retrieval? Just to exclude it is not related to a Cron job. An impatient user here. Thought it could be done based on push back update from Sonarr?

**Comment by bugsyb** [comment]
Found something... this is what now popped up in logs (names changed, but looks like proper names for episodes).
The name matches to what is seen at Ombi level and same as Sonarr level.
Something maybe is expecting a Series title and other is not returning it or the other way around?
```
2020-03-23 22:04:16.452 +01:00 [Debug] Batch acquisition of 0 triggers
2020-03-23 22:04:43.837 +01:00 [Debug] Batch acquisition of 1 triggers
2020-03-23 22:05:00.002 +01:00 [Debug] Batch acquisition of 0 triggers
2020-03-23 22:05:00.021 +01:00 [Debug] Calling Execute on job Emby.IEmbyContentSync
2020-03-23 22:05:00.596 +01:00 [Debug] Batch acquisition of 1 triggers
2020-03-23 22:05:00.599 +01:00 [Debug] Calling Execute on job Emby.IEmbyEpisodeSync
2020-03-23 22:05:00.599 +01:00 [Debug] Batch acquisition of 0 triggers
2020-03-23 22:05:00.600 +01:00 [Debug] Trigger instruction : NoInstruction
2020-03-23 22:05:01.668 +01:00 [Information] The episode "Name1" does not relate to a series, so we cannot save this
2020-03-23 22:05:01.672 +01:00 [Information] The episode "Name1" does not relate to a series, so we cannot save this
2020-03-23 22:05:01.676 +01:00 [Information] The episode "Name3" does not relate to a series, so we cannot save this
2020-03-23 22:05:01.838 +01:00 [Debug] Batch acquisition of 1 triggers
2020-03-23 22:05:01.838 +01:00 [Debug] Trigger instruction : DeleteTrigger
2020-03-23 22:05:01.838 +01:00 [Debug] Deleting trigger
2020-03-23 22:05:01.840 +01:00 [Debug] Batch acquisition of 0 triggers
2020-03-23 22:05:01.840 +01:00 [Debug] Calling Execute on job Emby.IEmbyAvaliabilityChecker
2020-03-23 22:05:01.917 +01:00 [Debug] Trigger instruction : DeleteTrigger
2020-03-23 22:05:01.917 +01:00 [Debug] Batch acquisition of 1 triggers
2020-03-23 22:05:01.917 +01:00 [Debug] Deleting trigger
2020-03-23 22:05:01.921 +01:00 [Debug] Calling Execute on job System.IRefreshMetadata
2020-03-23 22:05:01.922 +01:00 [Information] Starting the Metadata refresh
2020-03-23 22:05:01.924 +01:00 [Debug] Batch acquisition of 0 triggers
2020-03-23 22:05:01.981 +01:00 [Debug] Trigger instruction : DeleteTrigger
2020-03-23 22:05:01.982 +01:00 [Debug] Deleting trigger

```

**Comment by S1M0N** [comment]
I just want to hop in here and echo pretty much all of the above. I have checked my DB and systems along with the processes others have used and am seeing the same data overall. Using GoT as an example (as it seems to be consistent for most here) I have all episodes in Plex, Sonarr see this, but Ombi shows "Processing" on all episodes.

**Comment by aetaric** [comment]
To add in a bit, I'm running an outdated version (3.0.4817) and I do not have this issue. That might give @tidusjar and the other devs an idea of where/when this was introduced.

**Comment by Sjefen** [comment]
I just started using Ombi with Plex for a little while ago but almost gave up befor i saw this tread.
I Installed Ombi v3.0.4892 on Debian 10. Added my Plex server. All OK with movies, but not a singel TV Series are marked as available. So the problem must be between Ombi and Plex... (nothing to do with Sonar)
I installed after that Sonarr. Request are working but when they are downloaded they are not getting marked as Available when they are available on Plex.

**Comment by mattispantone81** [comment]
I’d like to add a note to this thread to advise that I’m having the same issues per above. 

I’m running...
- Sonnar v3.0.3.741
- Ombi v3.0.4892
- Unraid 6.8.1

All movies showing as available, but only a few TV Shows are showing as available. I’ve confirmed connections to Radarr and Sonarr are working ok for requests. Have performed manual sync. 

Log files attached

Sonarr Showing Big Bang is fully available...
![8DA16A38-2DE6-4682-8732-28978DB68EC4](https://user-images.githubusercontent.com/20289796/78949866-53922000-7b10-11ea-915d-f07e4efd5bed.png)

Ombi showing Big Bang is not available, but when viewing seasons is fully available...
![1134F8D4-BF8A-48DF-BD80-90024D573EA1](https://user-images.githubusercontent.com/20289796/78949907-7b818380-7b10-11ea-9c57-328fd2164a88.png)
[log-20200410.txt](https://github.com/tidusjar/Ombi/files/4459115/log-20200410.txt)

**Comment by damusmf** [comment]
Still having this issue.  Multiple TV Series are not showing as available and will have a request button next to them.  But if you try to request it then there is an error, "We already have this request... Please choose the select option for this request."  So i do that and ALL episodes are listed as "processing request" even though they exist in Sonarr and in Plex. Confirmed both Plex and Sonarr are successfully connected to Ombi, ran a manual full sync and can see it all processing with no errors.   I am using a combined MariaDB (MySQL) database for ombi. Radarr/Movies work with no problems. 

Will use Game of Thrones as the example here.  

![image](https://user-images.githubusercontent.com/31552122/79074552-b0c5d700-7cb2-11ea-81cd-1bf87640c67d.png)
![image](https://user-images.githubusercontent.com/31552122/79074566-c804c480-7cb2-11ea-9cff-89cc28851d2d.png)
![image](https://user-images.githubusercontent.com/31552122/79074636-26ca3e00-7cb3-11ea-9374-9a35bb584d8f.png)
![image](https://user-images.githubusercontent.com/31552122/79074643-2e89e280-7cb3-11ea-8ed0-e96e4b0d2915.png)

**Comment by Fiala06** [comment]
Having the same issue.
Unraid 6.8.3
OMBI 3.0.4892
Sonarr 2.0.0.5344




![image](https://user-images.githubusercontent.com/5183146/79254532-bb5ea880-7e39-11ea-9283-ebbdec4db63a.png)

![image](https://user-images.githubusercontent.com/5183146/79254571-c6b1d400-7e39-11ea-8946-4cba43e9fd1a.png)

![image](https://user-images.githubusercontent.com/5183146/79254664-ec3edd80-7e39-11ea-92e1-0eff941e3bf2.png)

**Comment by tidusjar** [comment]
Hi sorry but there has been no further progress on this yet.

Please remember that I have a full time (quite demanding) job, a toddler, a wife and a puppy. 
All of my limited free time is usually spent either developing Ombi, fixing bugs on Ombi or providing support with Ombi, keeping in mind that no one is paying me to do this.

Ombi is currently a 1 person development team. 

So if I have not yet got around to this you now know why 😄


But what would be useful is to include any logs, so run the sync on the Plex settings wait 5 minutes and then upload some log files, this should include information on whats happening when trying to sync the episodes.

**Comment by mattispantone81** [comment]
Thanks Jamie. You’re doing an amazing job. 

Did I not include logs? I thought I did. 



Regards,


Mat Fehlandt

**Comment by tidusjar** [comment]
Sorry @mattispantone81  You did at the very bottom of the message, I didn't notice it.
Thanks,

Looking through your logs everything looks ok, On ombi in your Big Bang example, if you Press Request -> Select ... and then click each season it will show you the availability per episode. There must be atleast one episode that is not available, can you find that and upload some more screenshots please?

**Comment by Fiala06** [comment]
> Sorry @mattispantone81 You did at the very bottom of the message, I didn't notice it.
> Thanks,
> 
> Looking through your logs everything looks ok, On ombi in your Big Bang example, if you Press Request -> Select ... and then click each season it will show you the availability per episode. There must be atleast one episode that is not available, can you find that and upload some more screenshots please?

99% resolved my issue after reading your response. Turns out Plex in ombi was no longer connected. I had to update the plex token and server-id. Once done did full sync noticed Big bang eps were officially saying available except for the very last episode even though it is. Getting closer! 

![image](https://user-images.githubusercontent.com/5183146/79358247-d262d080-7ef5-11ea-95cf-c0bd152a81fe.png)


![image](https://user-images.githubusercontent.com/5183146/79358287-dd1d6580-7ef5-11ea-8918-e995a8a8e6f8.png)

**Comment by Sjefen** [comment]
Attached a log file...
Movies as mentioned are OK but i have zero Series showing as Available

[log-20200415.txt](https://github.com/tidusjar/Ombi/files/4483497/log-20200415.txt)

**Comment by ShiroiKuma** [comment]
Debian 10.4
Ombi 3.0.4892
Plex

I'm having the same issue here where TV Shows requested prior to Ombi are not syncing, but those requested through Ombi are working fine.

In order to solve this myself and help Jamie potentially. I've done the following:

Logs show no errors, even with Debug turned on. I've deleted and rebuilt the .db files over and over. Still no luck. So in order to debug better, I've spent the last 2 days getting a Windows VM setup, python, VS2017/VS2019 in order to try and see why PlexEpidodes table only has 50 out of the expected 3571 records.

I copied my .db files to my VM, ran Ombi in Visual Studio 2019, in doing this it's working fine. On Windows 10 VM (master branch), it works perfectly. Not sure how to debug on Debian yet.

**Comment by goldenpipes** [comment]
This has been driving me nuts the last week or so as well @tidusjar 

as a test i removed a show from my plex server that i know i have 100% of, cleared ombiexternal, requested the show, had sonarr import, and have 100% of the episodes available, ran a recently added sync and nothing changed on ombi.

Ran a full sync and nothing changed.

Removed ombi from my plex - authorized devices, removed plex media server from ombi, deleted ombiexternal, restarted ombi, re-added plex, ran full sync, and voila the episodes were there.

I've done this 2 or 3 times and i can not figure out what is happened, its almost like Plex doesn't tell ombi there are new episodes even though there are, maybe a token is expiring or something?

**Comment by tscibilia** [comment]
@tidusjar I have the same issue as @Fiala06 is posting in [this comment](https://github.com/tidusjar/Ombi/issues/3282#issuecomment-614120664), however, I think I have a path to follow. 

That particular episode (and others that ombi seems to think is not available for me) happen to be 2-part episodes that only has 1 file. For example sonarr understands that the file "How I Met Your Mother - S07E23-E24 - The Magician's Code - Part One.m4v" applies to both ep 23 and ep24.

**Comment by Zoz92** [comment]
Same as all describe here, try the @goldenpipes solution but not working for me

**Comment by tonyrking** [comment]
I'll add to this, the same occurring for me also. Movies.. all fine. TV Shows showing nothing as available, only if you request it'll start downloading a fresh copy of the series again!!!

Scratching my head on this one.. most odd.

Sonarr V.3

Ombi can see the the folder of TV shows when you look in the Plex settings.
I'd have thought Ombi might discuss this with Plex to get what's there as well as Sonarr?

What a pickle.

**Comment by tonyrking** [comment]
ok.. think I've sussed it guys!!

Have you all put your plex username/password in the plex settings of Ombi?
Once I added that (after making sure tokens and everything else is filled out) saved.

I monitored the plex console .. you'll then see Ombi loging in and going through every episode. 
Let that finish, refresh your Ombi page.. and all the tv shows are showing as available. (request is showing for series that have a missing episode or two.. or ones that are two parters, which drives me crazy as usually the two parter is one double episode.. not sure how to fix that)

Good luck!!!

**Comment by tidusjar** [comment]
It's a known issue currently with double episodes

**Comment by tonyrking** [comment]
I think perhaps it should be something in Sonarr where you can 'link' two episode together as complete and whole.

**Comment by tidusjar** [comment]
Please upgrade to develop if you haven't already.

**Comment by tonyrking** [comment]
> Please upgrade to develop if you haven't already.

Of what? Sonarr?

**Comment by tidusjar** [comment]
Ombi, sorry I should have been a bit more specific

**Comment by Zoz92** [comment]
Will try and give u feedback :)

**Comment by Zoz92** [comment]
Version | 3.0.5164
use linuxserver/ombi:development
Working for me on rasp4 4gb Raspbian32 bits

**Comment by ndoty** [comment]
I switched from the main apt repo to the develop repo and it is working as expected now.

**Comment by Zoz92** [comment]
4.0.362.0
use  linuxserver/ombi:v4-preview
Working for me on rasp4 4gb Raspbian32 bits

And this new interface looks great :)

**Comment by TRI-JasonAnderosn** [comment]
> 
> 
> ok.. think I've sussed it guys!!
> 
> Have you all put your plex username/password in the plex settings of Ombi?
> Once I added that (after making sure tokens and everything else is filled out) saved.
> 
> Pressed..Manually run full sync.
> 
> I monitored the plex console .. you'll then see Ombi loging in and going through every episode.
> Let that finish, refresh your Ombi page.. and all the tv shows are showing as available. (request is showing for series that have a missing episode or two.. or ones that are two parters, which drives me crazy as usually the two parter is one double episode.. not sure how to fix that)
> 
> Good luck!!!

Ditto to this somewhat.  The last part of my troubleshooting was to question whether or not the plex token and machine auth token were wrong.  I've never put in my username/password for plex there, only the server info.  After putting in the login details the machine auth changed (and perhaps the plex token, I cant remember) and then I ran a full manual sync and everything started showing as available.

**Comment by lyz-code** [comment]
bump

**Comment by Duggie264** [comment]
I have a similar issue, but only with series that contain double episodes (e.g. NotEnemies S09E23-24 The One In A Tropical Island.mkv)

I note that tidusjar said this is a known issue with Double Episodes

All my other series have synced up fine, once I ensure I had a valid tokens and a full sync.

**Comment by carvitup** [comment]
> I have a similar issue, but only with series that contain double episodes (e.g. NotEnemies S09E23-24 The One In A Tropical Island.mkv)
> 
> I note that tidusjar said this is a known issue with Double Episodes
> 
> All my other series have synced up fine, once I ensure I had a valid tokens and a full sync.

I had come across this issue too and if you check plex it will only show Not Enemies S09E23 in this case, so Ombi thinks it is missing episode 24 from Plex. For Plex to see both episodes it needs to follow the format of S##E##-E## ( reference https://support.plex.tv/articles/naming-and-organizing-your-tv-show-files/ ) this corresponds in Sonarr > Settings > Media Management > Multi Episode Style > "Prefixed Range"

I had this working in v3 and it fixed the main issue for this thread but now that I am on v4 I seem to be having an issue with any shows syncing again - hoping to troubleshoot that too for tidusjar

**Comment by tidusjar** [comment]
Hey guys, I'm going to close this issue, there are mixed messages in here and multiple issues that are not directly related.

Please if you are having any issues can you please raise a new issue on github for this. I know this doesn't seem like the best approach, but it will be easier for me to see the 'wood through the trees' sort of speak.

Thank you

**Comment by hazemania** [comment]
> ok.. think I've sussed it guys!!
> 
> Have you all put your plex username/password in the plex settings of Ombi?
> Once I added that (after making sure tokens and everything else is filled out) saved.
> 
> Pressed..Manually run full sync.
> 
> I monitored the plex console .. you'll then see Ombi loging in and going through every episode.
> Let that finish, refresh your Ombi page.. and all the tv shows are showing as available. (request is showing for series that have a missing episode or two.. or ones that are two parters, which drives me crazy as usually the two parter is one double episode.. not sure how to fix that)
> 
> Good luck!!!

This solved it for me. Thanks.
